app/models/world_location_email_signup.rb

```diff
@@ -0,0 +1,52 @@
+class WorldLocationEmailSignup
+  attr_reader :feed
+
+  def initialize(feed)
+    @feed = feed
+  end
+
+  def signup_url
+    subscriber_list = Services.email_alert_api.find_or_create_subscriber_list(criteria)
+    subscriber_list["subscriber_list"]["subscription_url"]
+  end
+
+  def name
+    world_location.name
+  end
+
+  def valid?
+    uri && world_location_slug && world_location.present?
+  end
+
+private
+
+  def criteria
+    {
+      "links" => {
+        "world_locations" => [
+          world_location.content_id,
+        ],
+      },
+    }.merge("title" => world_location.name)
+  end
+
+  def world_location
+    @world_location ||= WorldLocation.find_by(slug: uri.path.match(%r{^/world/(.*)\.atom$})[1])
+  end
+
+  def uri
+    @uri ||= begin
+               URI.parse(feed)
+             rescue URI::InvalidURIError
+               nil
+             end
+  end
+
+  def world_location_slug
+    world_path = uri.path.match(%r{^/world/(.*)\.atom$})
+
+    return unless world_path
+
+    world_path[1]
```

Comment:
Minor - you could get this down to a one liner with `uri.path.match(%r{^/world/(.*)\.atom$}).try(:[], 1)`

Reply:
Nice one liner, saved 4 lines! 🔻